frontend/src/pages/register.jsx

```diff
@@ -19,126 +18,115 @@ const Heading = () => (
-      <input
-        id="address"
-        type="text"
-        {...register('address', { required: 'Address is required' })}
-        className="mt-2 px-3 py-2 w-full border rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
-      />
-      {errors.address && <p className="text-red-500 text-sm">{errors.address.message}</p>}
-    </div>
-
-    <div>
-      <label htmlFor="password" className="block text-lg font-medium text-gray-700">Password</label>
-      <input
-        id="password"
-        type="password"
-        {...register('password', { required: 'Password is required' })}
-        className="mt-2 px-3 py-2 w-full border rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
-      />
-      {errors.password && <p className="text-red-500 text-sm">{errors.password.message}</p>}
-    </div>
-
-    <button
-      type="submit"
-      className="w-full px-6 py-3 bg-blue-600 text-white font-bold rounded-lg hover:bg-blue-700 transition duration-300 ease-in-out"
-      disabled={isLoading}
-    >
-      {isLoading ? 'Creating account...' : 'Create Account'}
-    </button>
-  </form>
-);
-
-// Register: The main registration page
 const Register = () => {
   const { register, handleSubmit, reset, formState: { errors } } = useForm();
-  const [registerUser, { isLoading }] = useRegisterUserMutation();
+  const [registerUser, { isLoading }] = useRegisterUserMutation(); // Updated hook usage
   const navigate = useNavigate();
 
-  // Use useCallback to memoize the onSubmit handler
-  const onSubmitHandler = React.useCallback(async (data) => {
+  const onSubmit = async (data) => {
     try {
+      console.log(data);
       await registerUser(data).unwrap();
       reset();
       navigate('/login');
-    } catch (err) {
-      console.error('Registration failed', err);
+    } catch (error) {
+      console.error('Registration failed', error);
     }
-  }, [registerUser, reset, navigate]);
+  };
 
   return (
     <CenteredBox>
       <Heading />
-      <RegisterForm
-        onSubmit={onSubmitHandler}
-        register={register}
-        handleSubmit={handleSubmit}
-        errors={errors}
-        isLoading={isLoading}
-      />
+      <form className="mt-8 space-y-6" onSubmit={handleSubmit(onSubmit)}>
+        <div>
+          <label htmlFor="first_name" className="block text-lg font-medium text-gray-700">First Name</label>
+          <input
+            id="first_name"
+            type="text"
+            {...register('first_name', { required: 'First name is required' })}
+            className="mt-2 px-3 py-2 w-full border rounded-lg focus:outline-none focus:ring-2 focus:ring-blue-500"
+          />
+          {errors.first_name && <p className="text-red-500 text-sm">{errors.first_name.message}</p>}
+        </div>
+
+        <div>
+          <label htmlFor="last_name" className="block text-lg font-medium text-gray-700">Last Name</label>
```

Comment:
Ensure the required fields are correctly passed. Including re_password to avoid issues with the server